daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1482,6 +1518,74 @@ object Main {
         }
       }
 
+      case Some(conf.encodeEXI) => {
+        val encodeOpts = conf.encodeEXI
+        val channel = encodeOpts.output.toOption match {
+          case Some("-") | None => Channels.newChannel(STDOUT)
+          case Some(file) => new FileOutputStream(file).getChannel()
+        }
+        val output = Channels.newOutputStream(channel)
+
+        val inputStream = encodeOpts.infile.toOption match {
+          case Some("-") | None => STDIN
+          case Some(file) => {
+            val f = new File(file)
+            new FileInputStream(f)
+          }
+        }
+        val input = new InputSource(inputStream)
+
+        val exiFactory = {
+          if (encodeOpts.schema.isDefined)
+            getExiFactoryOpt(InfosetType.EXISA, encodeOpts.schema.toOption)
+          else
+            getExiFactoryOpt(InfosetType.EXI, encodeOpts.schema.toOption)
+        }
+        val exiResult = new EXIResult(exiFactory.get)
+        exiResult.setOutputStream(output)
+
+        val reader = XMLReaderFactory.createXMLReader()
+        reader.setContentHandler(exiResult.getHandler)
+        reader.parse(input)
+        output.close
```

Comment:
Do we need to catch any exceptions and return a exit code? Otherwise any exceptions will probably be reported as a bug, when really it should be reported as "invalid EXI/XML" or something.

Reply:
I would like to see tests which encode an invalid XML file and decode an invalid EXI file to make sure they get an error message, not a bug stack trace.  I'm sure people will try to encode an EXI file and decode an XML file the wrong way around and they should see an appropriate error message.